arches/app/models/card.py

```diff
@@ -154,35 +154,24 @@ def __init__(self, *args, **kwargs):
 
                 if "widgets" in args[0]:
                     for widget in args[0]["widgets"]:
-                        cardxnodexwidgetid = widget.get("id", None)
                         node_id = widget.get("node_id", None)
                         card_id = widget.get("card_id", None)
                         widget_id = widget.get("widget_id", None)
-                        if cardxnodexwidgetid is None and (
```

Comment:
Ah, thanks for finding the root cause. This fixes it for me without a throttle by deferring all the work to the db instead of using a retrieve-update-save pattern, which is where the potential for race conditions creeps in:

```diff
diff --git a/arches/app/models/card.py b/arches/app/models/card.py
index 57eb529dbe..44746350ca 100644
--- a/arches/app/models/card.py
+++ b/arches/app/models/card.py
@@ -163,10 +163,10 @@ class Card(models.CardModel):
                             and widget_id is not None
                         ):
                             widget_model, _ = (
-                                models.CardXNodeXWidget.objects.get_or_create(
+                                models.CardXNodeXWidget.objects.update_or_create(
                                     node_id=node_id,
                                     card_id=card_id,
-                                    widget_id=widget_id,
+                                    defaults={"widget_id": uuid.UUID(widget_id)},
                                 )
                             )
```

Reply:
Ah - you're right. That fixed the issue without the the throttle. Thanks!